cql3/statements/raw/describe_statement.hh

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright (C) 2022-present ScyllaDB
+ *
+ * Modified by ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: (AGPL-3.0-or-later and Apache-2.0)
+ */
+
+#pragma once
+
+#include "cql3/functions/function_name.hh"
+#include "cql3/statements/raw/parsed_statement.hh"
+#include "cql3/result_set.hh"
+#include "query-request.hh"
+#include "cql3/cf_name.hh"
+
+#include <memory>
+#include <seastar/core/sstring.hh>
+
+namespace cql3 {
+
+namespace statements {
+
+class prepared_statement;
+
+namespace raw {
+
+class describe_statement : public parsed_statement {
+private:
+    enum class describe_type {
+        cluster,
+        schema,
+        keyspace,
+        listing,
+        keyspaces_listing,
+        element,
+        generic
+    };
+
+    describe_type _desc_type;
+    sstring _element_type;
+    std::optional<sstring> _element_keyspace;
+    std::optional<sstring> _element_name;
+
+    bool _with_internals;
+    bool _is_listing;
+    bool _full_schema;
+    bool _only_keyspace;
+
+    ::shared_ptr<selection::selection> get_selection(schema_ptr schema);
+    query::partition_slice get_slice(schema_ptr schema);
+
+public:
+    inline static const sstring SCHEMA = "schema";
+    inline static const sstring KEYSPACE = "keyspace";
+    inline static const sstring TABLE = "table";
+    inline static const sstring INDEX = "index";
+    inline static const sstring VIEW = "view";
+    inline static const sstring TYPE = "type";
+    inline static const sstring FUNCTION = "function";
+    inline static const sstring AGGREGATE = "aggregate";
+
+    describe_statement(describe_type desc_type);
+    describe_statement(bool full_schema);
```

Comment:
Single-parameter constructors should be `explicit`. 

Use a `bool_class` for boolean parameters, otherwise it's hard to guess what a `describe_statement(false)` means.

Consider using a describe_statement_config struct instead (which can then be initialized with designated initializers, making the call site more readable).

Reply:
Created config and reduced constructors of `describe_statement` to 1 (that takes this config). Also using `bool_class`